### ISSUE TITLE
[bitcoin]: inscritpion offset mismatched 

### ISSUE THREAD

**Issue by popcnt1** [rooch-network/rooch#2441]
output of rooch statedb genesis-verifyt:

offset in data source(height is 85566(included)): 50228297

offset in statedb(rooch relay from  852202(included) to 855666) : 49643926

```
[inscription] mismatched: exp: "Inscription { txid: d8cf4fe99687989c52a2d69908b4e32c0f728fb0d2e1464c9bb74e0182e46975, index: 0, offset: 50228297, sequence_number: 447319, inscription_number: 447106, is_curse: false, body: [123, 34, 112, 34, 58, 34, 98, 114, 99, 45, 50, 48, 34, 44, 34, 111, 112, 34, 58, 34, 109, 105, 110, 116, 34, 44, 34, 116, 105, 99, 107, 34, 58, 34, 112, 117, 110, 107, 34, 44, 34, 97, 109, 116, 34, 58, 34, 49, 34, 125], content_encoding: MoveOption { vec: [] }, content_type: MoveOption { vec: [text/plain;charset=utf-8] }, metadata: [], metaprotocol: MoveOption { vec: [] }, parents: [], pointer: MoveOption { vec: [] }, rune: MoveOption { vec: [] } }", act: "Inscription { txid: d8cf4fe99687989c52a2d69908b4e32c0f728fb0d2e1464c9bb74e0182e46975, index: 0, offset: 49643926, sequence_number: 447319, inscription_number: 447106, is_curse: false, body: [123, 34, 112, 34, 58, 34, 98, 114, 99, 45, 50, 48, 34, 44, 34, 111, 112, 34, 58, 34, 109, 105, 110, 116, 34, 44, 34, 116, 105, 99, 107, 34, 58, 34, 112, 117, 110, 107, 34, 44, 34, 97, 109, 116, 34, 58, 34, 49, 34, 125], content_encoding: MoveOption { vec: [] }, content_type: MoveOption { vec: [text/plain;charset=utf-8] }, metadata: [], metaprotocol: MoveOption { vec: [] }, parents: [], pointer: MoveOption { vec: [] }, rune: MoveOption { vec: [] } }"
```

result in https://ordinals.com/inscription/447106 (2024/8/15 21:21 utc+8)

![image](https://github.com/user-attachments/assets/84d16278-f40a-47dc-8473-33108962100c)

**Comment by baichuan3** [comment]
1. The inscription #7569e482014eb79b4c46e1d2b08f720f2ce3b40899d6a2529c988796e94fcfd8i0 spent as gas fee at height **855412**, and final transfer at height **855629**. 

     Therefore, it is necessary to synchronize Rooch to the corresponding height before comparing the offset. In the above case, before Rooch synchronizes to the height **855412**,  both the offset and owner of Rooch and Bitcoin should be inconsistent.

- spent as gas fee tx https://ordiscan.com/tx/74b2c06b6a85b4d1897dcf14e7dbe9239e25bfe80f2c33734dbad19560e8dc58
- final transfer tx https://ordiscan.com/tx/df662337ec2622a3df79e2e708c063b9eb7bef3aa028bd8444b4eea9588ea0d6

2. Use this tool to view the records of inscription transfers, https://ordiscan.com/inscription/447106. Please note that some inscription data may be inaccurate sometimes.

**Comment by popcnt1** [comment]
@baichuan3 rooch has sync height at 855666, so it should be consistent

**Comment by jolestar** [comment]
The inscription result in rooch:
```bash
rooch rpc request --method btc_queryInscriptions --params '[{"inscription_id":{"txid":"7569e482014eb79b4c46e1d2b08f720f2ce3b40899d6a2529c988796e94fcfd8","index":0}},null,null,null]'
```
```json
{
  "data": [
    {
      "id": "0xd23c49fb9a742624498390a5b59c90e968c11d52b6437278a0e2eb9b82824103aa2a6f19511f00f4ebe2a5057cde8d78aff9e568becd4fb4df341ae9f0a4e63a",
      "owner": "rooch1fhey2uf6vpqm6fdmz4sxrscd4gltl90d5lthz5w4uqvq9l3ztkrqzl5nwm",
      "owner_bitcoin_address": "1Fu41MmbZLFHebUsDya8B3oqoG2MMZGiML",
      "flag": 0,
      "state_root": "0x5350415253455f4d45524b4c455f504c414345484f4c4445525f484153480000",
      "size": "0",
      "created_at": "1722815681000",
      "updated_at": "1722936378000",
      "object_type": "0x4::ord::Inscription",
      "value": {
        "txid": "0xd8cf4fe99687989c52a2d69908b4e32c0f728fb0d2e1464c9bb74e0182e46975",
        "bitcoin_txid": "7569e482014eb79b4c46e1d2b08f720f2ce3b40899d6a2529c988796e94fcfd8",
        "index": 0,
        "offset": "49643926",
        "sequence_number": 447319,
        "inscription_number": 447106,
        "is_curse": false,
        "body": "0x7b2270223a226272632d3230222c226f70223a226d696e74222c227469636b223a2270756e6b222c22616d74223a2231227d",
        "content_encoding": null,
        "content_type": "text/plain;charset=utf-8",
        "metadata": "0x",
        "metaprotocol": null,
        "parents": "",
        "pointer": null
      },
      "tx_order": "13656655",
      "state_index": "2101"
    }
  ],
  "next_cursor": {
    "tx_order": "13656655",
    "state_index": "2101"
  },
  "has_next_page": false
}
```

**Comment by jolestar** [comment]
should be resolved after #2500
